rs/pocket_ic_server/src/state_api/state.rs

```diff
@@ -740,6 +740,13 @@ impl ApiState {
         }
     }
 
+    pub async fn delete_all_instances(&self) {
+        let num_instances = self.instances.read().await.len();
+        for instance_id in 0..num_instances {
+            self.delete_instance(instance_id).await;
+        }
```

Comment:
this could be parallelized with https://docs.rs/tokio/latest/tokio/task/struct.JoinSet.html

Reply:
at least I think so. we have a read lock, and then a lock on the thing inside the instance vec only.

Reply:
Took a slight refactor: https://github.com/dfinity/ic/pull/2043/commits/f41a251b7f414e3045d4d32d9d0303c46b66ca58 Could you please take one more look?